plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/async/AsyncTransferManager.java

```diff
@@ -131,6 +131,10 @@ private void uploadInParts(
             .bucket(uploadRequest.getBucket())
             .key(uploadRequest.getKey())
             .overrideConfiguration(o -> o.addMetricPublisher(statsMetricPublisher.multipartUploadMetricCollector));
+
+        if (uploadRequest.getMetadata() != null && !uploadRequest.getMetadata().isEmpty()) {
```

Comment:
You can replace these repeated checks with `CollectionUtils.isEmpty(uploadRequest.getMetadata())`

Reply:
Thanks for the suggestion!